### PR TITLE
add curvefs recycle bin

### PR DESCRIPTION
**Pull request by cw123**
Signed-off-by: chenwei <461432360@qq.com>

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/fuse_client.cpp [hunk 1]
```diff
@@ -784,6 +769,271 @@ CURVEFS_ERROR FuseClient::RemoveNode(fuse_req_t req, fuse_ino_t parent,
...
+std::string GetRecycleTimeDirName() {
+    time_t timeStamp;
+    time(&timeStamp);
+    struct tm p = *localtime_r(&timeStamp, &p);
+    char now[64];
+    strftime(now, 64, "%Y-%m-%d-%H", &p);
+    return now;
+}
+
+CURVEFS_ERROR FuseClient::CreateManageNode(fuse_req_t req, uint64_t parent,
```
Comment (ilixiaocui) [inline curvefs/src/client/fuse_client.cpp:781]: What does ManagerNode mean?

File: curvefs/src/client/fuse_client.h [hunk 2]
```diff
@@ -296,11 +297,27 @@ class FuseClient {
  protected:
     CURVEFS_ERROR MakeNode(fuse_req_t req, fuse_ino_t parent, const char* name,
                            mode_t mode, FsFileType type, dev_t rdev,
-                           fuse_entry_param* e);
+                           bool internal, fuse_entry_param* e);
 
     CURVEFS_ERROR RemoveNode(fuse_req_t req, fuse_ino_t parent,
                              const char* name, FsFileType type);
 
+    CURVEFS_ERROR CreateManageNode(fuse_req_t req, uint64_t parent,
```
Comment (ilixiaocui) [inline curvefs/src/client/fuse_client.h:305]: Add notes here

File: curvefs/src/client/fuse_client.h [hunk 3]
```diff
@@ -296,11 +297,27 @@ class FuseClient {
  protected:
     CURVEFS_ERROR MakeNode(fuse_req_t req, fuse_ino_t parent, const char* name,
                            mode_t mode, FsFileType type, dev_t rdev,
-                           fuse_entry_param* e);
+                           bool internal, fuse_entry_param* e);
```
Comment (ilixiaocui) [inline curvefs/src/client/fuse_client.h:300]: What does internal flag mean?

File: curvefs/src/metaserver/recycle_cleaner.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ public:
+    explicit RecycleCleaner(const std::shared_ptr<Partition> &partition)
+        : partition_(partition) {
+        isStop_ = false;
+        LOG(INFO) << "RecycleCleaner poolId = " << partition->GetPoolId()
+                  << ", fsId = " << partition->GetFsId()
+                  << ", partitionId = " << partition->GetPartitionId();
+    }
+
+    void SetCopysetNode(copyset::CopysetNode *copysetNode) {
```
Comment (ilixiaocui) [inline curvefs/src/metaserver/recycle_cleaner.h:45]: Why copysetNode do not use shared_ptr

File: curvefs/src/metaserver/recycle_cleaner.h [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        copysetNode_ = copysetNode;
+    }
+
+    void SetMdsClient(std::shared_ptr<MdsClient> mdsClient) {
```
Comment (ilixiaocui) [inline curvefs/src/metaserver/recycle_cleaner.h:49]: Are SetXX for unit testing?

File: curvefs/src/metaserver/recycle_cleaner.h [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+#include <string>
+
+#include "curvefs/src/client/rpcclient/metaserver_client.h"
+#include "curvefs/src/metaserver/copyset/copyset_node_manager.h"
+#include "curvefs/src/metaserver/partition.h"
+
+namespace curvefs {
+namespace metaserver {
+using curvefs::client::rpcclient::MetaServerClient;
+class RecycleCleaner {
```
Comment (ilixiaocui) [inline curvefs/src/metaserver/recycle_cleaner.h:35]: Add a comment to this class

File: curvefs/proto/common.proto [hunk 7]
```diff
@@ -83,6 +83,7 @@ message PartitionInfo {
     optional uint64 inodeNum = 10;  // hearbeat upload this value to mds topo, topo/metaserver needn't persist this value
     optional uint64 dentryNum = 11;  // hearbeat upload this value to mds topo, topo/metaserver needn't persist this value
     map<int32, uint64> fileType2inodeNum = 12;
+    optional bool manageFlag = 13;
```
Comment (ilixiaocui) [inline curvefs/proto/common.proto:86]: add notes

File: curvefs/proto/mds.proto [hunk 8]
```diff
@@ -96,6 +98,7 @@ message FsInfo {
     required string owner = 12;
     optional uint64 txSequence = 13;
     optional string txOwner = 14;
+    optional uint32 recycleTimeHour =15;
```
Comment (ilixiaocui) [inline curvefs/proto/mds.proto:101]: add notes

File: curvefs/src/client/fuse_client.h (hunk 3 above)
Comment (cw123) [reply curvefs/src/client/fuse_client.h:300]: 区分请求来自外部（fuse）还是来自内部（回收站）

Distinguish between requests coming from outside (fuse) or from inside (recycle bin)

**Comment by cw123** [comment]
这个dest branch is abandoned

**Pull request closed without merging**